<!doctype html><html lang=en><head><meta charset=utf-8><title>How to Capture the Exception Message From Assert.Throws in NUnit | JM (Jason Meridth)</title><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Hugo 0.99.1"><meta name=viewport content="width=device-width,initial-scale=1"><meta name=keywords content=".net,testing"><meta name=description content="In the past I’ve used:
[ExpectedException(typeof(CustomException), ExpectedMessage=&#34;blah blah blah&#34;)] to set an exception expectation and assert on the …"><link rel=icon type=image/png href=/favicon-32x32.png sizes=32x32><link rel=icon type=image/png href=/favicon-16x16.png sizes=16x16><link rel=stylesheet type=text/css media=screen href=/css/normalize.css><link rel=stylesheet type=text/css media=screen href=/css/main.css><link rel=stylesheet type=text/css media=screen href=/css/all.css><link rel=stylesheet href=/css/katex.css crossorigin=anonymous><script defer src=/js/katex.js integrity=sha384-HELAAZU8xvHgfT/8z4Mhmu+E2z3oBrMEuywaMh/CEd5uTZIDSct7TEaX+S43+dOi crossorigin=anonymous></script>
<script defer src=/js/auto-render.js integrity=sha384-0QrOZ8WIARJUoT5pH7Iuzi7iq0mXUXLF9yj5ZvYxTkMRv1zcaPQiOQ9ksHYbKw7M crossorigin=anonymous onload=renderMathInElement(document.body)></script>
<script>document.addEventListener("DOMContentLoaded",function(){renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"$",right:"$",display:!1}]})})</script><script>localStorage.setItem("theme","dark")</script><meta property="og:title" content="How to Capture the Exception Message From Assert.Throws in NUnit"><meta property="og:description" content="In the past I’ve used:
[ExpectedException(typeof(CustomException), ExpectedMessage=&#34;blah blah blah&#34;)] to set an exception expectation and assert on the message. With the recent releases of NUnit I’ve finally transitioned all my code to use:
Assert.Throws<CustomException>(() => someType.someMethod(someValue)); or
Assert.Throws<CustomException>(someType.someMethod); if the method has no parameters.
Only lately have I needed to capture the exception message and validate it. To do this you do the following:
var ex = Assert.Throws<CustomException>(() => someType.someMethod(someValue));  Assert."><meta property="og:type" content="article"><meta property="og:url" content="http://blog.jasonmeridth.com/posts/how-to-capture-exception-message-from-assert-throws-in-nunit/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2012-03-23T00:00:00-05:00"><meta property="article:modified_time" content="2012-03-23T00:00:00-05:00"><meta name=twitter:card content="summary"><meta name=twitter:title content="How to Capture the Exception Message From Assert.Throws in NUnit"><meta name=twitter:description content="In the past I’ve used:
[ExpectedException(typeof(CustomException), ExpectedMessage=&#34;blah blah blah&#34;)] to set an exception expectation and assert on the message. With the recent releases of NUnit I’ve finally transitioned all my code to use:
Assert.Throws<CustomException>(() => someType.someMethod(someValue)); or
Assert.Throws<CustomException>(someType.someMethod); if the method has no parameters.
Only lately have I needed to capture the exception message and validate it. To do this you do the following:
var ex = Assert.Throws<CustomException>(() => someType.someMethod(someValue));  Assert."><meta itemprop=name content="How to Capture the Exception Message From Assert.Throws in NUnit"><meta itemprop=description content="In the past I’ve used:
[ExpectedException(typeof(CustomException), ExpectedMessage=&#34;blah blah blah&#34;)] to set an exception expectation and assert on the message. With the recent releases of NUnit I’ve finally transitioned all my code to use:
Assert.Throws<CustomException>(() => someType.someMethod(someValue)); or
Assert.Throws<CustomException>(someType.someMethod); if the method has no parameters.
Only lately have I needed to capture the exception message and validate it. To do this you do the following:
var ex = Assert.Throws<CustomException>(() => someType.someMethod(someValue));  Assert."><meta itemprop=datePublished content="2012-03-23T00:00:00-05:00"><meta itemprop=dateModified content="2012-03-23T00:00:00-05:00"><meta itemprop=wordCount content="94"><meta itemprop=keywords content=".net,testing,"></head><body><header><div id=avatar><a href=http://blog.jasonmeridth.com/><img src=/img/avatar.jpg alt="JM (Jason Meridth)"></a></div><div id=titletext><h2 id=title><a href=http://blog.jasonmeridth.com/>JM (Jason Meridth)</a></h2></div><div id=title-description><p id=subtitle>Learn, Converse, Share</p><div id=social><nav><ul><li><a href=https://github.com/jmeridth><i title=Github class="icons fab fa-github"></i></a></li><li><a href=mailto:jmeridth@gmail.com><i title=Email class="icons fas fa-envelope"></i></a></li><li><a href=https://linkedin.com/in/jmeridth><i title=LinkedIn class="icons fab fa-linkedin"></i></a></li><li><a href=/index.xml><i title=RSS class="icons fas fa-rss"></i></a></li><li><a><i title="Switch Dark Mode" class="dark-mode icons fas fa-moon"></i></a></li></ul></nav></div></div><div id=mainmenu><nav><ul><li><a href=/>Home</a></li><li><a href=/posts>All Posts</a></li><li><a href=/about>About</a></li><li><a href=/tags>Tags</a></li><li><a href=/categories>Categories</a></li></ul></nav></div></header><main><div class=post><article><div class=post-header><div class=meta><div class=date><span class=day>23</span>
<span class=rest>Mar 2012</span></div></div><div class=matter><h1 class=title>How to Capture the Exception Message From Assert.Throws in NUnit</h1><p class=post-meta><span class=post-meta>&nbsp;<i class="fas fa-clock"></i>&nbsp;1&nbsp;minutes
&nbsp;| &nbsp;
<i class="fas fa-book"></i>&nbsp;94&nbsp;words</span></p></div></div><div class=markdown><p>In the past I’ve used:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-csharp data-lang=csharp><span style=display:flex><span><span style=color:#a6e22e>[ExpectedException(typeof(CustomException), ExpectedMessage=&#34;blah blah blah&#34;)]</span>
</span></span></code></pre></div><p>to set an exception expectation and assert on the message. With the recent releases of NUnit I’ve finally transitioned all my code to use:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-csharp data-lang=csharp><span style=display:flex><span>Assert.Throws&lt;CustomException&gt;(() =&gt; someType.someMethod(someValue));
</span></span></code></pre></div><p>or</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-csharp data-lang=csharp><span style=display:flex><span>Assert.Throws&lt;CustomException&gt;(someType.someMethod);
</span></span></code></pre></div><p>if the method has no parameters.</p><p>Only lately have I needed to capture the exception message and validate it. To do this you do the following:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-csharp data-lang=csharp><span style=display:flex><span><span style=color:#66d9ef>var</span> ex = Assert.Throws&lt;CustomException&gt;(() =&gt; someType.someMethod(someValue));
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>Assert.That(ex.Message, Is.EqualTo(<span style=color:#e6db74>&#34;expected message here&#34;</span>));
</span></span></code></pre></div><p>I am blogging about this because I keep forgetting how to capture the exception message. Hope it helps someone else.</p></div><div class=tags><div class=taxosfloating_left><p>Tags</p></div><div class=termsfloating_right><p><a href=/tags/.net/>.net</a>
<a href=/tags/testing/>testing</a></p></div><div class=clearit></div></div></article></div></main><footer>© Copyright notice | <a href=https://github.com/dataCobra/hugo-vitae target=_blank>Vitae</a> theme for <a href=https://gohugo.io target=_blank>Hugo</a>
<script type=application/javascript>var doNotTrack=!1;doNotTrack||(window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-38456568-3","auto"),ga("send","pageview"))</script><script async src=https://www.google-analytics.com/analytics.js></script></footer><script src=/js/dark-mode.js></script></body></html>